node/src/p2p.rs

```diff
@@ -93,12 +128,106 @@ impl Service for P2p {
 
 #[async_trait]
 pub trait P2pService: Service<Args = P2pArgs, Command = P2pCmd, Error = P2pError> {
+    async fn wait_connected(&self) -> Result<()> {
+        let (tx, rx) = oneshot::channel();
+
+        self.send_command(P2pCmd::WaitConnected { respond_to: tx })
+            .await?;
+
+        Ok(rx.await?)
+    }
+
     async fn network_info(&self) -> Result<NetworkInfo> {
         let (tx, rx) = oneshot::channel();
 
         self.send_command(P2pCmd::NetworkInfo { respond_to: tx })
             .await?;
-        rx.await.map_err(|_| P2pError::WorkerDied)
+
+        Ok(rx.await?)
+    }
+
+    async fn exchange_header_request(&self, request: HeaderRequest) -> Result<Vec<ExtendedHeader>> {
+        if request.amount == 0 {
```

Comment:
doesn't it indicate some strange behaviour? I mean Err there?

Reply:
Like `InvalidAmount`?

Reply:
or just InvalidRequest?

Reply:
I just think if someone is making such request, should we indicate that or rather allow them?

Reply:
Celestia implementation does not return anything if we allow them and we get `OutboundFailure::Timeout` later on. So I prefer to either return empty Vec or error. Both are valid for me.